<a href="https://colab.research.google.com/github/YaduSM001/Datascience-assignment/blob/main/Data_Acquisition_Case_Study.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ***Task: Acquiring and Analyzing a Free Dataset from the Internet***

In [ ]:
import pandas as pd

In [ ]:
file_path='/content/goalscorers.csv'
df=pd.read_csv(file_path)
print(df.head())

         date  home_team away_team       team            scorer  minute  \
0  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene    44.0   
1  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín    55.0   
2  1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín    70.0   
3  1916-07-02      Chile   Uruguay    Uruguay   José Piendibene    75.0   
4  1916-07-06  Argentina     Chile  Argentina     Alberto Ohaco     2.0   

   own_goal  penalty  
0     False    False  
1     False    False  
2     False    False  
3     False    False  
4     False    False  


In [ ]:
file_path='/content/goalscorers.csv'
df=pd.read_csv(file_path)
print(df.head(4))

         date home_team away_team     team            scorer  minute  \
0  1916-07-02     Chile   Uruguay  Uruguay   José Piendibene    44.0   
1  1916-07-02     Chile   Uruguay  Uruguay  Isabelino Gradín    55.0   
2  1916-07-02     Chile   Uruguay  Uruguay  Isabelino Gradín    70.0   
3  1916-07-02     Chile   Uruguay  Uruguay   José Piendibene    75.0   

   own_goal  penalty  
0     False    False  
1     False    False  
2     False    False  
3     False    False  


In [ ]:
print('Statical summary')
print(df.describe)

Statical summary
<bound method NDFrame.describe of              date  home_team away_team       team            scorer  minute  \
0      1916-07-02      Chile   Uruguay    Uruguay   José Piendibene    44.0   
1      1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín    55.0   
2      1916-07-02      Chile   Uruguay    Uruguay  Isabelino Gradín    70.0   
3      1916-07-02      Chile   Uruguay    Uruguay   José Piendibene    75.0   
4      1916-07-06  Argentina     Chile  Argentina     Alberto Ohaco     2.0   
...           ...        ...       ...        ...               ...     ...   
44357  2024-07-13     Canada   Uruguay    Uruguay       Luis Suárez    90.0   
44358  2024-07-14  Argentina  Colombia  Argentina  Lautaro Martínez   112.0   
44359  2024-07-14      Spain   England      Spain     Nico Williams    47.0   
44360  2024-07-14      Spain   England    England       Cole Palmer    73.0   
44361  2024-07-14      Spain   England      Spain   Mikel Oyarzabal    86.0   



***1.What is the size of the dataset? (Number of rows and columns)***

In [ ]:
df.shape

(44362, 8)

The above output shows that how much rows and columns are there in the dataset

***2: What are the data types of each column?***

In [ ]:
df.dtypes

date          object
home_team     object
away_team     object
team          object
scorer        object
minute       float64
own_goal        bool
penalty         bool
dtype: object

The above result shows the data type on each column that is present in the dataset

**3: Are there any missing values in the dataset? If so, which columns have
missing values and how many?**

In [ ]:
missing_values=df.isnull().sum()
missing_columns = missing_values[missing_values > 0]
missing_columns

scorer     49
minute    259
dtype: int64

The above result shows that there are 49 missing values in scorer and 259 missing values in minute

 ***4: What are the unique values in a categorical column of your choice? How
many unique values are there?***

In [ ]:
column_name = 'home_team'
unique_values = df[column_name].unique()
num_unique_values = len(unique_values)
print("unique_values")
print(unique_values)

print("\num_unique_values")
print(num_unique_values)

unique_values
['Chile' 'Argentina' 'Brazil' 'Uruguay' 'Paraguay' 'Czechoslovakia'
 'Italy' 'Switzerland' 'United States' 'Hungary' 'France' 'Netherlands'
 'Republic of Ireland' 'Egypt' 'Sweden' 'Bolivia' 'Peru' 'Belgium'
 'Portugal' 'Germany' 'Spain' 'Latvia' 'Estonia' 'Lithuania' 'Yugoslavia'
 'Poland' 'Haiti' 'Mexico' 'Luxembourg' 'Bulgaria' 'Israel' 'Austria'
 'Romania' 'Finland' 'Norway' 'Greece' 'Cuba' 'Ecuador' 'Colombia'
 'Northern Ireland' 'Wales' 'Scotland' 'England' 'Turkey' 'Japan'
 'Saarland' 'South Korea' 'Hong Kong' 'Guatemala' 'Sudan' 'Costa Rica'
 'Indonesia' 'Denmark' 'German DR' 'Syria' 'China PR' 'Canada' 'Russia'
 'Curaçao' 'Iceland' 'Ethiopia' 'Ghana' 'Honduras' 'Nigeria' 'Suriname'
 'Taiwan' 'Morocco' 'Cyprus' 'Tunisia' 'Malta' 'El Salvador' 'Jamaica'
 'Nicaragua' 'Panama' 'Albania' 'India' 'Trinidad and Tobago' 'Venezuela'
 'DR Congo' 'Ivory Coast' 'North Korea' 'Australia' 'Algeria' 'Congo'
 'Iran' 'Myanmar' 'Zambia' 'Bermuda' 'Cameroon' 'Senegal' 'Libya'
 'Zimb

The output shows that there are 220 unique values in the home_team

***5: For a numerical column of your choice, calculate the mean, median, and
standard deviation***

In [ ]:
column_name = 'own_goal'
mean_value = df[column_name].mean()
median_value = df[column_name].median()
std_deviation = df[column_name].std()

mean_value, median_value, std_deviation

(0.018506830169965287, 0.0, 0.13477661842161154)

***6: Identify any potential outliers in a numerical column of your choice. Explain
your approach.***

In [ ]:
column_name = 'minute'
Q1 = df[column_name].quantile(0.25)
Q3 = df[column_name].quantile(0.75)
IQR = Q3 - Q1
lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR
outliers = df[(df[column_name] < lower_bound) | (df[column_name] > upper_bound)]
outliers

Empty DataFrame
Columns: [date, home_team, away_team, team, scorer, minute, own_goal, penalty]
Index: []

The Interquartile Range (IQR) method is used to identify outliers, where any data point below 𝑄1−1.5×𝐼𝑄𝑅Q1−1.5×IQR or above 𝑄3+1.5×𝐼𝑄𝑅Q3+1.5×IQR is considered a potential outlier, with 𝑄1Q1 and 𝑄3Q3 representing the 25th and 75th percentiles

***7: Find the correlation between numerical columns and discuss any interesting
relationships.***

In [ ]:
df_numeric = df.apply(pd.to_numeric, errors='coerce')
df_numeric = df_numeric.dropna()
correlation_matrix = df_numeric.corr()

print("Correlation matrix:")
correlation_matrix

Correlation matrix:


date  home_team  away_team  team  scorer  minute  own_goal  \
date        NaN        NaN        NaN   NaN     NaN     NaN       NaN   
home_team   NaN        NaN        NaN   NaN     NaN     NaN       NaN   
away_team   NaN        NaN        NaN   NaN     NaN     NaN       NaN   
team        NaN        NaN        NaN   NaN     NaN     NaN       NaN   
scorer      NaN        NaN        NaN   NaN     NaN     NaN       NaN   
minute      NaN        NaN        NaN   NaN     NaN     NaN       NaN   
own_goal    NaN        NaN        NaN   NaN     NaN     NaN       NaN   
penalty     NaN        NaN        NaN   NaN     NaN     NaN       NaN   
year        NaN        NaN        NaN   NaN     NaN     NaN       NaN   
month       NaN        NaN        NaN   NaN     NaN     NaN       NaN   
day         NaN        NaN        NaN   NaN     NaN     NaN       NaN   

           penalty  year  month  day  
date           NaN   NaN    NaN  NaN  
home_team      NaN   NaN    NaN  NaN  
away_team      NaN   NaN    NaN  NaN  
team           NaN   NaN    NaN  NaN  
scorer         NaN   NaN    NaN  NaN  
minute         NaN   NaN    NaN  NaN  
own_goal       NaN   NaN    NaN  NaN  
penalty        NaN   NaN    NaN  NaN  
year           NaN   NaN    NaN  NaN  
month          NaN   NaN    NaN  NaN  
day            NaN   NaN    NaN  NaN

***8: Group the data by a categorical column and compute the aggregate statistics
(e.g., mean, sum) for a numerical column.***

In [ ]:
categorical_column = 'scorer'
numerical_column = 'minute'

grouped_data = df.groupby(categorical_column)[numerical_column].agg(['mean', 'sum'])

grouped_data

mean    sum
scorer                            
A'ala Hubail      55.153846  717.0
A. Elangovan      42.000000   42.0
Aage Rou Jensen   85.000000   85.0
Aaran Lines       56.666667  170.0
Aaron Boupendza   16.000000   16.0
...                     ...    ...
Žarko Tomašević   58.250000  233.0
Željko Milinovič  23.500000   47.0
Željko Čajkovski  61.500000  369.0
Ștefan Dobay      58.400000  292.0
Ștefan Iovan      54.000000  162.0

[14335 rows x 2 columns]